base/src/AffineTransform.cpp

```diff
@@ -0,0 +1,380 @@
+include "AffineTransform.h"
+#include "FrameMetadata.h"
+#include "Frame.h"
+#include "Logger.h"
+#include "Utils.h"
+#include "AIPExceptions.h"
+#include "DMAFDWrapper.h"
+#include "math.h"
+#include "opencv2/core.hpp"
+#include "npp.h"
+#include "CuCtxSynchronize.h"
+
+#define PI 3.14159265
+
+class AffineTransform::Detail
+{
+public:
+	Detail(AffineTransformProps &_props) : props(_props), shiftX(0), shiftY(0), mFrameType(FrameMetadata::GENERAL), mFrameLength(0)
+	{
+		nppStreamCtx.hStream = props.stream->getCudaStream();
+	}
+
+	~Detail()
+	{
+	}
+
+	void setMetadata(framemetadata_sp &metadata)
+	{
+		if (mFrameType != metadata->getFrameType())
+		{
+			mFrameType = metadata->getFrameType();
+			switch (mFrameType)
+			{
+			case FrameMetadata::RAW_IMAGE:
+				mOutputMetadata = framemetadata_sp(new RawImageMetadata(FrameMetadata::MemType::DMABUF));
+				break;
+			default:
+				throw AIPException(AIP_FATAL, "Unsupported frameType<" + std::to_string(mFrameType) + ">");
+			}
+		}
+
+		if (!metadata->isSet())
+		{
+			return;
+		}
+
+		ImageMetadata::ImageType imageType = ImageMetadata::MONO;
+		if (mFrameType == FrameMetadata::RAW_IMAGE)
```

Comment:
Why is this if statement required? I think the switch case above is already taking care of this